### PR TITLE
Catch2 Unit Test Fixes

### PR DESCRIPTION
**Pull request by pvelesko**
* Kernel launches are not being error checked. This can lead to cases where a failure in kernel invocation is classified as a failure in some other operation
* `hipLaunchKernelGGL` does not return `hipError_t` so the check must be done via `hipGetLastError` 
* Block size of 512 exceeds the HW capabilities of some (integrated) GPUs
* `make_hipPitchedPtr` is passed x, x, y as arguments instead of x, y, z

### CODE REVIEW

**Comment by Kerilk** [comment]
Tagging @gargrahul as I forgot to mention this PR during our meeting.

**Review by mangupta — CHANGES_REQUESTED** [review]
@pvelesko / @Kerilk : Thank you for the PR. This PR has some merge conflicts. Can you please resolve them so that I can merge this in?

**Comment by pvelesko** [comment]
@mangupta done

File: tests/catch/unit/stream/hipMultiStream.cc [hunk 1]
```diff
@@ -49,13 +49,16 @@ TEST_CASE("Unit_hipMultiStream_sameDevice") {
...
+    HIP_CHECK(hipGetLastError());
     hipLaunchKernelGGL(HIP_KERNEL_NAME(nKernel), dim3(1), dim3(1), 0, 0, yd);
+    HIP_CHECK(hipGetLastError());
     HIP_CHECK(hipFree(data[i]));
     HIP_CHECK(hipStreamDestroy(streams[i]));
   }
   HIP_CHECK(hipMemcpy(&x, xd, sizeof(float), hipMemcpyDeviceToHost));
   HIP_CHECK(hipMemcpy(&y, yd, sizeof(float), hipMemcpyDeviceToHost));
   REQUIRE(x == Approx(y));
+  for (auto& stream : streams) HIP_CHECK(hipStreamDestroy(stream));
```
Comment (mangupta) [inline tests/catch/unit/stream/hipMultiStream.cc]: Aren't we already destroying the streams inside the for loop above? We would get an error for invalid stream on both AMD and Nvidia platform here causing the test to fail.

**Review by mangupta — CHANGES_REQUESTED** [review]

File: tests/catch/unit/stream/hipMultiStream.cc (hunk 1 above)
Comment (pvelesko) [reply tests/catch/unit/stream/hipMultiStream.cc]: Made a mistake during merge

**Comment by pvelesko** [comment]
@mangupta

**Comment by pvelesko** [comment]
@mangupta @Kerilk @gargrahul

**Comment by gargrahul** [comment]
@pvelesko  Could you please resubmit the required changes on top of the latest develop branch as a new PR ?

**Pull request closed without merging**